### PR TITLE
Add a task retrigger feature

### PR DESCRIPTION
**Pull request by lissyx**

### CODE REVIEW

**Comment by petemoore** [comment]
You may want to strip some routes from the original task, etc. I think the tools site has some details about what mutations we make when running tasks. Looks like a nice first stab though!

File: cmds/task/actors.go [hunk 1]
```diff
@@ -41,6 +43,54 @@ func runRerun(credentials *tcclient.Credentials, args []string, out io.Writer, _
...
+	// TaskDefinitionResponse: https://github.com/taskcluster/taskcluster-client-go/blob/88cfe471bfe2eb8fc9bc22d9cde6a65e74a9f3e5/tcqueue/types.go#L1554-L1716
+	newT.Dependencies  = t.Dependencies
+	newT.Extra         = t.Extra
+	newT.Metadata      = t.Metadata
+	newT.Payload       = t.Payload
+	newT.Priority      = t.Priority
+	newT.ProvisionerID = t.ProvisionerID
+	newT.Requires      = t.Requires
+	newT.Retries       = t.Retries
+	newT.Routes        = t.Routes
```
Comment (jonasfj) [inline cmds/task/actors.go]: I would strip routes if I where you...

Comment (jonasfj) [reply cmds/task/actors.go]: Routes can cause side effects like indexing...

File: cmds/task/actors.go [hunk 2]
```diff
@@ -41,6 +43,54 @@ func runRerun(credentials *tcclient.Credentials, args []string, out io.Writer, _
...
+	newT.SchedulerID   = t.SchedulerID
+	newT.Scopes        = t.Scopes
+	newT.Tags          = t.Tags
```
Comment (jonasfj) [inline cmds/task/actors.go]: Take a look at how it's done in taskcluster-tools...

I'm pretty sure there is an informal convention for adding either a tag or task.extra property pointing to the taskId it was retriggered from...

File: cmds/task/actors.go [hunk 3]
```diff
@@ -41,6 +43,54 @@ func runRerun(credentials *tcclient.Credentials, args []string, out io.Writer, _
...
+	newT.TaskGroupID   = t.TaskGroupID
+	newT.WorkerType    = t.WorkerType
```
Comment (petemoore) [inline cmds/task/actors.go]: nit: you could put these in the struct literal, i.e.

```go
	newT := &queue.TaskDefinitionRequest{
		SchedulerID: t.SchedulerID,
		Created:  tcclient.Time(now),
		Deadline: tcclient.Time(now.Add(24 * time.Hour)),
		Expires:  tcclient.Time(now.Add(24*time.Hour).AddDate(0, 1, 0)),
		Dependencies:  t.Dependencies,
		Extra:  t.Extra,
		.....
	}
```

File: cmds/task/actors.go [hunk 4]
```diff
@@ -41,6 +43,54 @@ func runRerun(credentials *tcclient.Credentials, args []string, out io.Writer, _
...
+
+	c, err := q.CreateTask(newTaskID, newT)
+	if err != nil {
+		return fmt.Errorf("could not create task: %v", err)
+	}
+
+	// If we got no error, that means the task was successfully rund
```
Comment (petemoore) [inline cmds/task/actors.go]: s/rund/submitted/

File: cmds/task/actors.go (hunk 3 above)
Comment (lissyx) [reply cmds/task/actors.go]: Thanks, I tried that but it was not working. Likely go-newbies issue :)

**Comment by lissyx** [comment]
Dustin gave me a link to that earlier: https://github.com/taskcluster/taskcluster-tools/blob/e8b6d45f10e7520f717b7a9f5db87d550c74d15e/src/views/UnifiedInspector/ActionsMenu.jsx#L141-L155

**Comment by djmitche** [comment]
Yeah, tools does not appear to strip the routes.  Stripping routes is probably most relevant to treeherder -- tasks without treeherder routes will not show up there.  I have a mild preference that the two act the same, but no real preference for whether that is with or without routes.

**Comment by petemoore** [comment]
CI failure is the same in both travis and taskcluster:

```
--- FAIL: TestLogCommand (1.03s)
	<autogenerated>:12: 
                          
	Error Trace:	fetchers_test.go:152
		
	Error:      	Not equal: 
		
	            	expected: "[taskcluster 2017-03-03 21:18:34.946Z] Task ID: TtAsnXdCS1-tAQxvMO4rHQ\n[taskcluster 2017-03-03 21:18:34.946Z] Worker ID: i-035dd1bd8da876f13\n[taskcluster 2017-03-03 21:18:34.946Z] Worker Group: us-west-1b\n[taskcluster 2017-03-03 21:18:34.946Z] Worker Node Type: m3.large\n[taskcluster 2017-03-03 21:18:34.946Z] Worker Type: tutorial\n[taskcluster 2017-03-03 21:18:34.946Z] Public IP: 54.153.13.193\n\n[taskcluster 2017-03-03 21:18:48.518Z] === Task Starting ===\nhello World\n[taskcluster 2017-03-03 21:18:48.945Z] === Task Finished ===\n[taskcluster 2017-03-03 21:18:48.946Z] Successful task run with exit code: 0 completed in 14.001 seconds\n"
		
	            	received: "{\n  \"code\": \"ResourceNotFound\",\n  \"message\": \"Artifact not found\\n----\\nmethod:     getLatestArtifact\\nerrorCode:  ResourceNotFound\\nstatusCode: 404\\ntime:       2018-04-27T11:27:28.993Z\",\n  \"requestInfo\": {\n    \"method\": \"getLatestArtifact\",\n    \"params\": {\n      \"0\": \"public/logs/live.log\",\n      \"taskId\": \"TtAsnXdCS1-tAQxvMO4rHQ\",\n      \"name\": \"public/logs/live.log\"\n    },\n    \"payload\": {},\n    \"time\": \"2018-04-27T11:27:28.993Z\"\n  }\n}\n"
		
	Messages:   	Log's are not equal.
```

I suspect it is unrelated, but I haven't looked into what that test is doing.

**Pull request merged**